### DISCUSSION TITLE
Possible to up- and download Files programmaticly

### DISCUSSION THREAD

**Discussion by jackserippi** [upscayl/upscayl#625] [Q&A] [answered]
First of all such a great software, awsome. I wonder if it is possible or a workaround available to upload files via an API, upscayl them and download the files afterwards, so the software could run in autopilot mode?

**Comment by NayamAmarshe** [comment, marked as answer]
We're currently building Upscayl Cloud for the same use-case :)

**Reply by jackserippi** [reply]
Great! Thought about that. Will the code for this also be available for community?

**Reply by NayamAmarshe** [reply]
> Great! Thought about that. Will the code for this also be available for community?

Upscayl Cloud cannot be self hosted and is a very modular service and since most cloud services require proprietary configuration anyway, Upscayl Cloud will remain a simple online service.

The desktop version will always be free and FOSS though, and will continue to get updates and new features :)

**Reply by RiasGFirst** [reply]
Will upsacyl cloud have a free version or a paid version

**Reply by NayamAmarshe** [reply]
We have to provide both, like any other service. The servers are quite expensive though, so Upscayl Cloud needs a sustainable model.
